### PR TITLE
Re-enable perf build; upgrade bcc

### LINKED ISSUE #3299: `perf` build fails with Alpine 3.9 tool chain
Trying to compile the perf utility the new alpine 3.9 based tool chain (see https://github.com/linuxkit/linuxkit/pull/3297) results in errors, such as:
```
  CC       builtin-bench.o
In file included from /usr/include/linux/byteorder/little_endian.h:13,
                 from /usr/include/asm/byteorder.h:5,
                 from /linux/tools/include/uapi/linux/perf_event.h:20,
                 from perf.h:8,
                 from builtin-bench.c:18:
/usr/include/linux/swab.h:161:8: error: unknown type name '__always_inline'
 static __always_inline __u16 __swab16p(const __u16 *p)
        ^~~~~~~~~~~~~~~
/usr/include/linux/swab.h:161:30: error: expected '=', ',', ';', 'asm' or '__attribute__' before '__swab16p'
 static __always_inline __u16 __swab16p(const __u16 *p)
                              ^~~~~~~~~
/usr/include/linux/swab.h:174:8: error: unknown type name '__always_inline'
```

See also https://github.com/linuxkit/linuxkit/issues/3296

### PR DESCRIPTION
**Pull request by kmjohansen**
**- What I did**

- Re-enable the perf build for 4.19.x and 5.3.x.

- Move bcc build up to 0.11.0 release and ensure that bcc programs build/run again

**- How I did it**
The 4.19.x kernel requires a pair of patches to support perf and bcc.  The first overcomes build failures from missing defines that were dropped in the 4.19 series but fixed in subsequent stable releases.  The second modifies the uapi swab.h file so that the unknown symbol errors from a missing __always_inline are resolved.  The second patch appears to be an outstanding issue that was never resolved in mainline.  The patch itself is sourced from lkml verbatim and rebased against 4.19 and 5.3

**- How to verify it**

- Build and run perf

- Build and run bcc

**- Description for the changelog**
Update perf and bcc builds. Fixes #3299

### CODE REVIEW

**Comment by rn** [comment]
Thanks for the PR. Do you know what the upstream status of the kernel patches is? I did find them on LKML but did not see any followup. I'm hesitant to carry forward kernel patches which are not upstream or won't make it upstream.

**Comment by kmjohansen** [comment]
The perf symbols patch was merged, and is present in 5.0 releases. AFAICT, the stable maintainers simply forgot, or didn’t realize it was necessary, to pull that patch back to the other stable releases.

For the swab patch, I’m less certain. It looks like the author’s patch was ignored and never resent. I didn’t see any discussion associated with it, tbh.

I understand the desire to carry as few patches as possible. This seems to be a case where we won’t get working diagnostic tools (bcc, and potentially bpftrace) without carrying this one line patch. From my perspective, the risk of not being able to diagnose performance problems outweighs the risk of diverging from upstream just a little bit.

File: kernel/Makefile [hunk 1]
```diff
@@ -149,9 +149,9 @@ show-tags: show-tag_$(2)$(3)$(4)
 # with DCT as part of the dependency on build_$(2)$(3)$(4) and then build
 # with DOCKER_CONTENT_TRUST explicitly set to 0
 
-# Skip perf build for now. See:
-# https://github.com/linuxkit/linuxkit/issues/3299
-ifeq ($(ARCH),SKIP)
+Only build perf on latest LTS and stable kernels.  Skip arm64.
```
Comment (rn) [inline kernel/Makefile]: This line needs commenting out.

**Comment by rn** [comment]
@kmjohansen I was going to pick up this PR, but the Makefile changes has an issue. Could you force push an update? (and apologies for the long delay, work/travel/holidays go in the way)

File: kernel/Makefile (hunk 1 above)
Comment (kmjohansen) [reply kernel/Makefile]: Thanks for catching this.  I've fixedup and re-pushed as requested.

**Comment by kmjohansen** [comment]
@rn No apology necessary.  Thanks for giving it another look.

File: kernel/Makefile [hunk 2]
```diff
@@ -149,9 +149,9 @@ show-tags: show-tag_$(2)$(3)$(4)
 # with DCT as part of the dependency on build_$(2)$(3)$(4) and then build
 # with DOCKER_CONTENT_TRUST explicitly set to 0
 
-# Skip perf build for now. See:
-# https://github.com/linuxkit/linuxkit/issues/3299
-ifeq ($(ARCH),SKIP)
+# Only build perf on latest LTS and stable kernels.  Skip arm64.
+ifeq ($(2), $(filter $(2),5.3.x 4.19.x))
+ifneq ($(ARCH),aarch64)
```
Comment (zlim) [inline kernel/Makefile]: Are there known issues on arm64 build?

Comment (kmjohansen) [reply kernel/Makefile]: I don't have an arm64 rig to test on, so I can't comment.  I'm merely reverting the fix that disabled `perf` entirely.  Prior to that, it was disabled for the arm64 build.  See here: https://github.com/linuxkit/linuxkit/commit/94cdfb55b62fbeb4829ad9575116d4d29e26186a#diff-74d64d85ef940cb8fbcd3540a1f97c9c

Comment (rn) [reply kernel/Makefile]: @zlim I can't remember if I originally tried on arm64 or not, but don't have the cycles at the moment to test bcc on arm64.

**Pull request merged**